auth-source-pass.el

```diff
@@ -293,28 +311,13 @@ components, from longest to shortest."
   (cl-maplist (lambda (components) (mapconcat #'identity components "."))
               name-components))
 
-(defun auth-source-pass--name-port-user-suffixes (name user port)
-  "Return a list of possible path suffixes for NAME, USER, & PORT.
-
-The resulting list is ordered from most specific to least
-specific, with paths matching all of NAME, USER, & PORT first,
-then NAME & USER, then NAME & PORT, then just NAME."
-  (seq-mapcat
-   #'identity
-   (list
-    (when (and user port)
-      (list
-       (format "%s@%s%s%s" user name auth-source-pass-port-separator port)
-       (format "%s%s%s/%s" name auth-source-pass-port-separator port user)))
-    (when user
-      (list
-       (format "%s@%s" user name)
-       (format "%s/%s" name user)))
-    (when port
-      (list
-       (format "%s%s%s" name auth-source-pass-port-separator port)))
-    (list
-     (format "%s" name)))))
+(defun sort-list-by-length (list)
```

Comment:
All functions and variables should start with the package-name. Also, the package name should be followed by 2 dashes to indicate that this function is private to the package. So, this function should be named `auth-source-pass--sort-list-by-length`.

You may want to activate [package-lint](https://github.com/purcell/package-lint) to check for similar errors automatically.